app/Http/Controllers/Order/OrderController.php

```diff
@@ -216,14 +225,48 @@ public function show($id)
             $allowDomainStatus = StatusSetting::pluck('domain_check')->first();
 
             return view('themes.default1.order.show',
-                compact('invoiceItems', 'invoice', 'user', 'order', 'subscription', 'licenseStatus', 'installationDetails', 'allowDomainStatus', 'noOfAllowedInstallation', 'getInstallPreference', 'currenctVersion', 'lastActivity'));
+                compact('invoiceItems', 'user', 'order', 'subscription', 'licenseStatus', 'installationDetails', 'allowDomainStatus', 'noOfAllowedInstallation', 'getInstallPreference', 'lastActivity','connectionLabel','versionLabel','date','licdate','supdate'));
         } catch (\Exception $ex) {
+            dd($ex);
             Bugsnag::notifyException($ex);
 
             return redirect()->back()->with('fails', $ex->getMessage());
         }
     }
 
+    public function getExpiryLabel($expiryDate)
+    {
+        if($expiryDate < (new Carbon())->toDateTimeString()) {
+             return getDateHtml($expiryDate)."<br>&nbsp;&nbsp;&nbsp;&nbsp;<span class='label label-danger' style='padding-left:4px;' <label data-toggle='tooltip' style='font-weight:500;' data-placement='top' title='Order has Expired'>
+                         </label>
+             <i class='fa fa-exclamation'></i>&nbsp;Expired</span>";
+        } else{
+            return getDateHtml($expiryDate);
+        }
+    }
+
+
+    public function getConnectionLabel($lastConnectionDate) 
+    {
+       return $lastConnectionDate->toDateTimeString() > (new Carbon('-30 days'))->toDateTimeString() ? "<br><span class='label label-primary' <label data-toggle='tooltip' style='font-weight:500;' data-placement='top' title='Installation is Active'>
+                         </label><i class= 'fa fa-check'></i>&nbsp;Active</span>": "<br><span class='label label-info' <label data-toggle='tooltip' style='font-weight:500;background-color:crimson;' data-placement='top' title='Installation inactive for more than 30 days'>
+                        </label> <i class= 'fa fa-info-circle'></i>&nbsp;Inactive</span>";
+    }
+
+
+    public function getLatestVersionLabel($currenctVersion,$productId)
+    {
+        $latestVersion = ProductUpload::where('product_id', $productId)->latest()->value('version');
```

Comment:
please use cache for these kinds of operations. See `agentAgentTimezone` in helpers for reference

Reply:
Fixed